pkg/controller/namespace/namespace_controller.go

```diff
@@ -129,19 +130,22 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
-	// initialize subresource if created by kubesphere
-	if workspace := namespace.Labels[tenantv1alpha1.WorkspaceLabel]; workspace != "" {
+	// Initialize subresource if created by kubesphere
+	if _, ok := namespace.Labels[tenantv1alpha1.WorkspaceLabel]; ok {
 		if err := r.bindWorkspace(rootCtx, logger, namespace); err != nil {
 			return ctrl.Result{}, err
 		}
-		if err := r.initRoles(rootCtx, logger, namespace); err != nil {
-			return ctrl.Result{}, err
-		}
 	} else {
 		if err := r.unbindWorkspace(rootCtx, logger, namespace); err != nil {
 			return ctrl.Result{}, err
 		}
 	}
+	// Need to init roles for devops/project namespaces
+	if _, ok := namespace.Labels[constants.NamespaceLabelKey]; ok {
```

Comment:
`NamespaceLabelKey` will be present in all namespaces. We need to distinguish between namesapce created through ks-console and namespace created directly through kubectl, which can be judged by `WorkspaceLabelKey` and `DevOpsProjectLabelKey`or create builtin roles for all namespaces.

Reply:
@wansir, maybe we can change to following
```
	_, hasWorkspaceLabel := namespace.Labels[tenantv1alpha1.WorkspaceLabel]
	_, hasDevOpsProjectLabel := namespace.Labels[constants.DevOpsProjectLabelKey]
	if hasWorkspaceLabel || hasDevOpsProjectLabel {
```

Reply:
Agreed, this should be part of initRoles.